sdk/monitor/azure-monitor-query/src/main/java/com/azure/monitor/query/implementation/logs/models/LogsQueryHelper.java

```diff
@@ -0,0 +1,153 @@
+     * @throws IllegalArgumentException if an instance of the object cannot be created.
+     */
+    public static <T> List<T> toObject(LogsTable table, Class<T> type) {
+        List<T> result = new ArrayList<>();
+        for (LogsTableRow tableRow : table.getRows()) {
+            try {
+                T t = type.newInstance();
+
+                Map<String, Field> declaredFieldMapping = Arrays.stream(type.getDeclaredFields())
+                        .collect(Collectors.toMap(field -> field.getName().toLowerCase(Locale.ROOT), field -> field));
+
+                tableRow.getRow().stream()
+                        .forEach(tableCell -> {
+                            String columnName = tableCell.getColumnName();
+                            try {
+                                Field field = declaredFieldMapping.get(columnName.toLowerCase(Locale.ROOT));
+                                if (field == null) {
+                                    return;
+                                }
+                                field.setAccessible(true);
+                                if (tableCell.getColumnType() == LogsColumnType.BOOL) {
+                                    field.set(t, tableCell.getValueAsBoolean());
+                                } else if (tableCell.getColumnType() == LogsColumnType.DATETIME) {
+                                    field.set(t, tableCell.getValueAsDateTime());
+                                } else if (tableCell.getColumnType() == LogsColumnType.DYNAMIC) {
+                                    if (tableCell.getValueAsDynamic() != null) {
+                                        field.set(t,
+                                                tableCell.getValueAsDynamic()
+                                                        .toObject(TypeReference.createInstance(field.getType())));
+                                    }
+                                } else if (tableCell.getColumnType() == LogsColumnType.INT) {
+                                    field.set(t, tableCell.getValueAsInteger());
+                                } else if (tableCell.getColumnType() == LogsColumnType.LONG) {
+                                    field.set(t, tableCell.getValueAsLong());
+                                } else if (tableCell.getColumnType() == LogsColumnType.REAL
+                                        || tableCell.getColumnType() == LogsColumnType.DECIMAL) {
+                                    field.set(t, tableCell.getValueAsDouble());
+                                } else if (tableCell.getColumnType() == LogsColumnType.STRING
+                                        || tableCell.getColumnType() == LogsColumnType.GUID
+                                        || tableCell.getColumnType() == LogsColumnType.TIMESPAN) {
+                                    field.set(t, tableCell.getValueAsString());
+                                }
+                                field.setAccessible(false);
```

Comment:
Should this not be in the finally block to make sure we do not accidently keep a field accessible?

Reply:
This doesn't permanently make the field accessible. It applies to only the current field instance and setting this back to false is strictly not required since I don't use this field instance anywhere else after this line.